internal/graph/cached_resolver.go

```diff
@@ -173,13 +173,13 @@ func (c *CachedCheckResolver) ResolveCheck(
 	return resp, nil
 }
 
-// checkRequestCacheKey converts the ResolveCheckRequest into a canonical cache key that can be
+// CheckRequestCacheKey converts the ResolveCheckRequest into a canonical cache key that can be
 // used for Check resolution cache key lookups in a stable way.
 //
 // For one store and model ID, the same tuple provided with the same contextual tuples and context
 // should produce the same cache key. Contextual tuple order and context parameter order is ignored,
 // only the contents are compared.
-func checkRequestCacheKey(req *ResolveCheckRequest) (string, error) {
+func CheckRequestCacheKey(req *ResolveCheckRequest) (string, error) {
```

Comment:
It's probably time that this moves out into a shared package. We have the `internal/keys` package that we've started to build out for shared request key logic. I would recommend we create a package under that for this shared stuff.

Reply:
I don't disagree but this function is tightly-coupled with the ResolveCheckRequest and will create a circular dependency if moved so will require a slightly more involved refactor. Are you ok with me deferring this work?

Reply:
Yep, that's fine. That's add an issue for it and move on in the scope of this PR.